dashboard/src/components/AppView/AppView.tsx

```diff
@@ -146,6 +146,9 @@ export default function AppView({
     setResourceRefs(parseResources(parsedManifest, cluster, app.namespace));
   }, [app, cluster]);
 
+  if (error && error.constructor === FetchError) {
+    return <Alert theme="danger">Application not found. Received: {error.message}</Alert>;
+  }
```

Comment:
This was an error I saw in several places, where the components was partially rendered with wrong information from the previous namespace. To avoid that, early return if there is an error fetching resources.

Reply:
Yes, it would be more clean if the state doesn't contain wrong info (in this case I believe is the component state, not the redux state the one containing old info, need to check) but we need this early return anyway. If not, we would render an empty (or loading gif) header and the error in a subcomponent, rather than at the top.